cypress/GenerateCypressTestSpecs.js

```diff
@@ -16,55 +16,82 @@
+
+function WatchAndRun(pathToWatch, functionToRun) {
+  var timeoutHandle;
+  var timeOfLastChange;
+  
+  fs.watch(pathToWatch, (eventType, filename) => {
+    var now = new Date();
+    if (timeOfLastChange && now - timeOfLastChange < 1000) { clearTimeout(timeoutHandle); }
+    timeOfLastChange = now;
+    timeoutHandle = setTimeout(functionToRun, 1000)
+  });
 }
-catch(error) { LogToFile(`Caught an Exception:\r\n${error}`)}
-fs.closeSync(logFile);
+
+// ----------------------------------------------------------------------
 
 function main() {
-  // Get the test list file that we MIGHT need to modify.
-  const pathToTestList = './cypress/TestLists.js';
-  var testList_jsContents = fs.readFileSync(pathToTestList, 'ascii');
-  // console.log(testList_jsContents)
-  // console.log()
-
-  // Get the list of test js files.
-  var testFileDirectory = fs.readdirSync(pathToTestFiles);
-  // console.log(testFileDirectory);
-  // console.log()
-
-  // Parse each test js file.
-  testFileDirectory.forEach(fileName => { ParseTestJsFile(fileName); });
-
-  // Finalize our findings by re-writing out the TestList.js file ONLY if it has changed.
-  var index = testList_jsContents.indexOf('// ************ Generated Code Beyond this Point *************************');
-  var newFileContents =
-    '// ************ Generated Code Beyond this Point *************************\r\n' +
-    '// Do NOT manually alter this file from this point onwards.\r\n' +
-    '// Any changes you make will be overridden at runtime.\r\n' +
-    'export const masterListOfAllTestCases = [\r\n';
-  fullTestList.forEach(testSpecification => { newFileContents += `  '${testSpecification}',\r\n` });
-  newFileContents += '];\r\n';
-
-  // Only write the file out if something has changed.
-  if (!testList_jsContents.endsWith(newFileContents)) {
-    newFileContents = testList_jsContents.substring(0, index) + newFileContents;
-    console.log(`NEW FILE CONTENTS ARE:\r\n${newFileContents}`)
-    fs.writeFileSync(pathToTestList, newFileContents)
+  try {
+    fullTestList = []
+
+    // Get the test list file that we MIGHT need to modify.
+    const pathToTestList = './cypress/TestLists.js';
+    var testList_jsContents = fs.readFileSync(pathToTestList, 'ascii');
+    // console.log(testList_jsContents)
+    // console.log()
+
+    // Get the list of test js files.
+    var testFileDirectory = fs.readdirSync(pathToTestFiles);
+    // console.log(testFileDirectory);
+    // console.log()
+
+    // Parse each test js file.
+    testFileDirectory.forEach(fileName => { ParseTestJsFile(fileName); });
+
+    // Finalize our findings by re-writing out the TestList.js file ONLY if it has changed.
+    var index = testList_jsContents.indexOf('// ************ Generated Code Beyond this Point *************************');
+    var newFileContents =
+      '// ************ Generated Code Beyond this Point *************************\r\n' +
```

Comment:
I had left comment in previous PR about using multiline strings. What was your opinion?
https://github.com/Microsoft/ConversationLearner-UI/pull/953#discussion_r251131899

Reply:
Liked the suggestion, made the fix.